components/selection/README.md

```diff
@@ -2,146 +2,328 @@
 
-**Events:**
+### Events
 
 | Event | Description |
-|--|--|
+|---|---|
 | `d2l-selection-action-click` | Dispatched when the user clicks the action button. The `SelectionInfo` is provided as the event `detail`. If `requires-selection` was specified then the event will only be dispatched if items are selected. |
+<!-- docs: end hidden content -->
 
-## d2l-selection-input
+## Selection Input [d2l-selection-input]
 
 The `d2l-selection-input` is a required component in selection controls - without it, there wouldn't be anything for the user to select! Note: `d2l-list-item` already provides a selection input for selectable list items. If `d2l-selection-input` is placed within a selection control that specifies `selection-single`, then radios will be rendered instead of checkboxes.
 
-**Properties:**
+<!-- docs: demo live name:d2l-selection-input display:block -->
+```html
+<script type="module">
+  import '@brightspace-ui/core/components/selection/selection-input.js';
+  import '@brightspace-ui/core/components/selection/demo/demo-selection.js';
+</script>
+<!-- docs: start hidden content -->
+<style>
+  ul {
+    padding: 0;
+  }
+  li {
+    list-style-type: none;
+    align-items: center;
+    display: flex;
+  }
+  d2l-selection-input {
+    margin-right: 10px;
+  }
+</style>
+<!-- docs: end hidden content -->
+<d2l-demo-selection>
+  <ul>
+    <li><d2l-selection-input key="geo" label="Geography" selected></d2l-selection-input>Geography</li>
+    <li><d2l-selection-input key="sci" label="Science"></d2l-selection-input>Science</li>
+  </ul>
+</d2l-demo-selection>
+```
+
+<!-- docs: start hidden content -->
+### Properties
 
 | Property | Type | Description |
-|--|--|--|
-| `key` | String, required | Key to identify the the selectable. |
+|---|---|---|
+| `key` | String, required | Key to identify the selectable. |
 | `label` | String | Accessible hidden label for the input. |
 | `labelled-by` | String | Id reference to the accessible label for the input. **Note:** if specified, it must reference an element in the same DOM context. |
 | `selected` | Boolean | State of the input. |
 
-**Accessibility:** either `label` or `labelled-by` is required.
-
-**Events:**
+### Events
 
 | Event | Description |
-|--|--|
+|---|---|
 | `d2l-selection-change` | Dispatched when the state of the input changes. |
+<!-- docs: end hidden content -->
 
-## d2l-selection-select-all
+### Accessibility Properties
+
+Either `label` or `labelled-by` is **required**.
+
+## Select All [d2l-selection-select-all]
 
 The `d2l-selection-select-all` is an optional component that provides a checkbox for bulk selecting the selectable elements within the selection control. Its state will also be automatically updated when the state of the selectable elements change.
 
-**Properties:**
+In the example below, setting `selection-for` to `other-list` demonstrates the ability to use `d2l-selection-select-all` for `d2l-select-input` outside of the custom selection element.
```

Comment:
I thought it could be cool to have 2 lists in order to demonstrate `selection-for`